apis/ec2/v1alpha1/custom_types.go

```diff
@@ -449,3 +449,132 @@ type CustomTransitGatewayRouteTableParameters struct {
+	// +optional
+	ResourceType *string `json:"resourceType"`
+	// The ID of the VPC for which you want to create a flow log.
+	// +optional
+	// +crossplane:generate:reference:type=github.com/crossplane-contrib/provider-aws/apis/ec2/v1beta1.VPC
+	VPCID *string `json:"vpcID,omitempty"`
+	// VPCIDRef is a reference to an API used to set
+	// the VPCID.
+	// +optional
+	VPCIDRef *xpv1.Reference `json:"vpcIDRef,omitempty"`
+	// VPCIDSelector selects references to API used
+	// to set the VPCID.
+	// +optional
+	VPCIDSelector *xpv1.Selector `json:"vpcIDSelector,omitempty"`
+
+	// The ID of the TransitGateway for which you want to create a flow log.
+	// +optional
+	// +crossplane:generate:reference:type=TransitGateway
+	TransitGatewayID *string `json:"transitGatewayId,omitempty"`
+
+	// TransitGatewayIDRef is a reference to an API used to set
+	// the TransitGatewayID.
+	// +optional
+	TransitGatewayIDRef *xpv1.Reference `json:"transitGatewayIdRef,omitempty"`
+
+	// TransitGatewayIDSelector selects references to API used
+	// to set the TransitGatewayID.
+	// +optional
+	TransitGatewayIDSelector *xpv1.Selector `json:"transitGatewayIdSelector,omitempty"`
+
+	// The ID of the TransitGatewayAttachment for which you want to create a flow log.
+	// +optional
+	// +crossplane:generate:reference:type=TransitGatewayVPCAttachment
+	TransitGatewayAttachmentID *string `json:"transitGatewayAttachmentId,omitempty"`
+
+	// TransitGatewayAttachmentIDRef is a reference to an API used to set
+	// the TransitGatewayAttachmentID.
+	// +optional
+	TransitGatewayAttachmentIDRef *xpv1.Reference `json:"transitGatewayAttachmentIdRef,omitempty"`
+
+	// TransitGatewayAttachmentIDSelector selects references to API used
+	// to set the TransitGatewayAttachmentID.
+	// +optional
+	TransitGatewayAttachmentIDSelector *xpv1.Selector `json:"transitGatewayAttachmentIdSelector,omitempty"`
+
+	// The ID of the one or more Subnets for which you want to create a flow log.
+	// +optional
+	// +crossplane:generate:reference:type=github.com/crossplane-contrib/provider-aws/apis/ec2/v1beta1.Subnet
+	SubnetID *string `json:"subnetId,omitempty"`
+
+	// SubnetIDRefs is a referenc to a SubnetID used to set
+	// the SubnetID.
+	// +optional
+	SubnetIDRef *xpv1.Reference `json:"subnetIdRef,omitempty"`
+
+	// SubnetIDSelector selects reference to a SubnetID used
+	// to set the SubnetID.
+	// +optional
+	SubnetIDSelector *xpv1.Selector `json:"subnetIdSelector,omitempty"`
+
+	// The ID of the NetworkInterface for which you want to create a flow log.
+	// +optional
+	NetworkInterfaceID *string `json:"networkInterfaceId"`
+
+	// The Amazon Resource Names (ARNs) of an IAM Role.
+	// +crossplane:generate:reference:type=github.com/crossplane-contrib/provider-aws/apis/iam/v1beta1.Role
+	// +crossplane:generate:reference:extractor=github.com/crossplane-contrib/provider-aws/apis/iam/v1beta1.RoleARN()
+	DeliverLogsPermissionARN *string `json:"deliverLogsPermissionARN,omitempty"`
```

Comment:
`json:"deliverLogsPermissionArn,omitempty"`

Reply:
Updated the json property